x-pack/test/detection_engine_api_integration/utils.ts

```diff
@@ -864,21 +883,27 @@ export const getRule = async (
 };
 
 /**
- * Waits for the rule in find status to be succeeded before continuing
+ * Waits for the rule in find status to be 'succeeded'
+ * or the provided status, before continuing
  * @param supertest Deps
  */
-export const waitForRuleSuccess = async (
+export const waitForRuleSuccessOrStatus = async (
   supertest: SuperTest<supertestAsPromised.Test>,
-  id: string
+  id: string,
+  status?: 'succeeded' | 'failed' | 'partial failure'
 ): Promise<void> => {
   // wait for Task Manager to finish executing the rule
+  let statusToWaitFor = status;
+  if (statusToWaitFor == null) {
+    statusToWaitFor = 'succeeded';
```

Comment:
This can be provided as a default value for `status` in the function parameter list instead of checking it inside the function